src/scenechange/fast.rs

```diff
@@ -86,7 +69,36 @@ impl<T: Pixel> SceneChangeDetector<T> {
           backward_adjusted_cost: delta as f64,
         }
       } else {
-        unreachable!()
+        // SAFETY: `frame_buffer` is unconditionally initialized at the start of this code block,
+        // and this branch is only reached when `frame_buffer` is `None`.
+        unsafe { std::hint::unreachable_unchecked() }
```

Comment:
I have mixed feelings about using this, although it does seem like we have it in a few places in ASM functions. It would actually be _really cool_ if there was a version of this that performed the check and panicked in debug mode, but ignored the check and allowed optimizations in release mode.

Edit: Actually it looks like such a crate exists: https://crates.io/crates/new_debug_unreachable

Reply:
Should we replace all usages of `unreachable_unchecked` with that macro then?

Reply:
I've replaced all usages of `unreachable_unchecked` with the macro from the crate you linked since there seems to be no downside. Thanks!

Reply:
Thanks! (I checked the macro code and I don't know any reason it shouldn't work :sweat_smile: )